src/textual/layouts/vertical.py

```diff
@@ -40,14 +38,11 @@ def arrange(
 
         y = box_models[0].margin.top if box_models else 0
 
-        displayed_children = parent.displayed_children
-
+        displayed_children = cast("list[Widget]", parent.displayed_children)
```

Comment:
just adding a cast, so displayed_children is not considered as just a list of DOMNodes

Reply:
I suspect DOMNodes and Widgets are going to merge at some point. The distinction was never clear.